DEVEL.md

```diff
@@ -0,0 +1,78 @@
+
+## Building the binary and the container image
+
+To build the Operator container image, run:
+```bash
+IMAGE=docker.io/<your username>/istio-operator:latest tmp/build/docker_build.sh
+```
+
+This builds the binary, downloads and patches the Istio Helm charts, and finally builds the container image. The build script doesn't push the image to the container registry, so you'll need to push it manually:
+```bash
+docker push docker.io/<your username>/istio-operator:latest
+```
+
+## Running the operator locally
+
+During development, you can run Istio Operator locally on your dev machine and have it reconcile objects in a local or remote OpenShift cluster. You can run the Operator from within your IDE and even use a debugger.
+
+To run the Operator locally, you need to set the following environment variables:
+
+- `KUBECONFIG=<path to your kubectl config file>`
+- `POD_NAME=istio-operator`
+- `POD_NAMESPACE=istio-operator`
+- `ISTIO_CNI_IMAGE=some-registry/istio-cni-image:latest`
+- `ISTIO_CNI_IMAGE_PULL_SECRET=my-pull-secret` (required if the registry hosting the CNI image is not public; secret must be in operator's namespace as defined above)
+
+Make sure you run `tmp/build/docker_build.sh` before running the Operator to generate the Helm charts in `tmp/_output/helm`.
+
+Before running the Operator, don't forget to stop the Operator running in the cluster:
+```bash
+oc -n istio-operator scale deployment istio-operator --replicas 0
+```  
+
+Run the operator locally with the following command-line options:
+```bash
+--resourceDir tmp/_output/resources/ --discoveryCacheDir /tmp/istio-operator/.kube/cache/discovery istio-operator
+```
+
+### Modifying the ValidatingWebhookConfiguration 
+
+Because the Operator installs a `ValidatingWebhookConfiguration` that points to the Operator pod running inside the cluster, all CRUD operations on `ServiceMeshControlPlane` or `ServiceMeshMemberRoll` objects will fail. To fix this, you either need to remove the `ValidatingWebhookConfiguration`:
+```bash
+oc delete validatingwebhookconfigurations istio-operator.servicemesh-resources.maistra.io 
+```
+
+or point it to the Operator process running on your local machine by patching the `Service` and `Endpoints` objects:
+```bash
+oc -n istio-operator patch svc admission-controller --type=json -p='[{"op": "remove", "path": "/spec/selector"}]'
+oc -n istio-operator patch ep admission-controller --type=json -p='[{"op": "add","path": "/subsets","value": [{"addresses": [{"ip": "192.168.1.100"}],"ports": [{"port": 11999}]}]}]'
+```
+
+Replace `192.168.1.100` with the IP of your dev machine, where the Operator is running. You also need to ensure that the OpenShift master node can connect to your dev machine on port 11999. Note: this is trivial if running OpenShift through DIND.
+
+ 
+NOTE: you'll need to remove the `ValidatingWebhookConfiguration` or patch the `Service` and `Endpoints` every time you restart the Operator, as it resets them during startup.
```

Comment:
What about putting the word `after` e.g.:

NOTE: You need to remove the `ValidatingWebhookConfiguration` or patch the `Service` and `Endpoints` every time after you restart the Operator, as it is created automatically during startup.

Reply:
done.